statsmodels/tsa/statespace/tests/test_simulate.py

```diff
@@ -1626,6 +1626,28 @@ def test_pandas_univariate_dateindex_repetitions():
     assert_(actual.columns.equals(desired.columns))
 
 
+def test_pandas_univariate_integer_index():
+    warning_text = (
+        "An unsupported integer index was provided and will be "
+        "converted to a range index with the same values."
+    )
+    index = pd.Index([i for i in range(3, 15)], dtype=int)
+    y = pd.DataFrame(data={0: np.arange(12)}, index=index)
+
+    # Fit the forecaster and ensure that the right warning is raised.
+    with pytest.warns(ValueWarning, match=warning_text):
+        forecaster = sarimax.SARIMAX(endog=y, k_factors=1, factor_order=1)
+        forecaster = forecaster.fit()
```

Comment:
Using `fit` can be problematic in tests (due to numerical instability across platforms or due to the time it takes to perform the fitting), so it's better to just use a fixed set of parameters, e.g. something like:

```python
forecaster = sarimax.SARIMAX(endog=y, order=(1, 0, 0))
forecaster = forecaster.smooth([0.5, 1.0])
```